src/Optimization/hiopResidual.cpp

```diff
@@ -320,6 +330,10 @@ int hiopResidual::update(const hiopIterate& it,
   int ierr = MPI_Allreduce(aux, aux_g, 6, MPI_DOUBLE, MPI_MAX, nlp->get_comm()); assert(MPI_SUCCESS==ierr);
   nrmInf_nlp_optim=aux_g[0]; nrmInf_nlp_feasib=aux_g[1]; nrmInf_nlp_complem=aux_g[2];
   nrmInf_bar_optim=aux_g[3]; nrmInf_bar_feasib=aux_g[4]; nrmInf_bar_complem=aux_g[5];
+  
+//  double sum_one_norm = 0.0;
+//  ierr = MPI_Allreduce(&nrmOne_theta, &sum_one_norm, 1, MPI_DOUBLE, MPI_SUM, nlp->get_comm()); assert(MPI_SUCCESS==ierr);
+//  nrmOne_theta = sum_one_norm;
```

Comment:
Constraints c and d are duplicated in each MPI process, and hence this MPI_Allreduce is not required.
Otherwise this will multiply the value of `nrmOne_theta` by the number of MPI processes.

This brings the issue that the current implementation of [hiopVectorPar::one_norm](https://github.com/LLNL/hiop/blob/master/src/LinAlg/hiopVectorPar.cpp#L304) may not provide the value we want, i.e., the computation is correct for the vector on the distributed `x` space, but wrong on the other spaces where we duplicates the local data.
Should we file another issue about it? Or just leave a comment before the implementation?

Reply:
if the MPI reduce in `hiopResidual::compute_nlp_norms` does not work for all NLP formulations, we should remove it and rely instead on non-local   ` hiopVectorPar::one_norm` as your implementation of  `hiopResidual::compute_nlp_norms` does right now. 

 ` hiopVectorPar::one_norm` should work for replicated `c` and `d` though ... if it does not, then these `hiopVectorPar` vectors are not created correctly, that are not constructed with MPI_COMM_SELF ... if they are, everything should work fine